*** ../latest/jgpcmodel/car.h	2011-06-30 15:22:01.000000000 -0400
--- car.h	2011-07-03 08:59:00.358462874 -0400
***************
*** 83,89 ****
      counter = 0;
      speed =0.0;
      steer = 0.0;
!     showWaypoints = false;
      steering = false;
      ready = 0;
      out = NULL;};
--- 83,89 ----
      counter = 0;
      speed =0.0;
      steer = 0.0;
!     //showWaypoints = false;
      steering = false;
      ready = 0;
      out = NULL;};
***************
*** 105,114 ****
    float adjustSpeed(float spd) {speed += spd; return speed;}; //modify speed
    float setSpeed(float spd) {speed = spd; return speed;}; // absolute speed set 
    // change speed return speed
    float adjustCourse(float turn) {steer = turn; return steer;}; 
    void setBodyColor(int, int); // color of body part
    float getHeading();  // returns current heading
!   float getDestinationHeading(); // returns heading of vector from current pos to destination
    void gatherPositionAndAttitude(double *);
    float steer2heading(float); // steers toward parameter returns current heading
    void recordTrack(FILE *ofile) {out = ofile;};
--- 105,115 ----
    float adjustSpeed(float spd) {speed += spd; return speed;}; //modify speed
    float setSpeed(float spd) {speed = spd; return speed;}; // absolute speed set 
    // change speed return speed
+   float getSpeed() {return speed;};
    float adjustCourse(float turn) {steer = turn; return steer;}; 
    void setBodyColor(int, int); // color of body part
    float getHeading();  // returns current heading
!   //float getDestinationHeading(); // returns heading of vector from current pos to destination
    void gatherPositionAndAttitude(double *);
    float steer2heading(float); // steers toward parameter returns current heading
    void recordTrack(FILE *ofile) {out = ofile;};
***************
*** 121,128 ****
    int bodyPartCount;
    int numCommands;
    int nextCommand;
!   int numWaypoints;
!   int wpcounter;
    int ready;
    float wradius; // wheel radius
    dBodyID body[MAXBODYPARTS+4];
--- 122,129 ----
    int bodyPartCount;
    int numCommands;
    int nextCommand;
!   //int numWaypoints;
!   //int wpcounter;
    int ready;
    float wradius; // wheel radius
    dBodyID body[MAXBODYPARTS+4];
***************
*** 133,143 ****
    dGeomID sphere[4];
    int color[MAXBODYPARTS];
    CMDS cmds[MAXCOMMANDS];
!   P2D waypoints[MAXWAYPOINTS];
!   P2D destination, currentpos, prevpos;
    int subcnt,counter;
    float speed, steer,adjsteer;
!   bool showWaypoints;
    bool steering;
    FILE *out;
    float YoffsetFromGoal(P2D,P2D,float);
--- 134,144 ----
    dGeomID sphere[4];
    int color[MAXBODYPARTS];
    CMDS cmds[MAXCOMMANDS];
!   //P2D waypoints[MAXWAYPOINTS];
!   //P2D destination, currentpos, prevpos;
    int subcnt,counter;
    float speed, steer,adjsteer;
!   //bool showWaypoints;
    bool steering;
    FILE *out;
    float YoffsetFromGoal(P2D,P2D,float);
