*** ../latest/jgpcmodel/onecar.cpp	2011-07-01 11:10:38.000000000 -0400
--- onecar.cpp	2011-07-02 12:08:10.972159450 -0400
***************
*** 157,162 ****
--- 157,163 ----
    }
    */
    // test for space collisions (cars are defined as spaces) test is more efficient
+   /*
    if (dGeomIsSpace (o1) || dGeomIsSpace (o2)) { 
      // colliding a space with something :
      dSpaceCollide2 (o1, o2, data,&nearCallback); 
***************
*** 167,173 ****
      if (dGeomIsSpace (o2))
        dSpaceCollide ((dSpaceID)o2, data, &nearCallback);
      
!   } else {
      n = dCollide (o1,o2,N,&contact[0].geom,sizeof(dContact));
      if (n > 0) {
        for (i=0; i<n; i++) {
--- 168,174 ----
      if (dGeomIsSpace (o2))
        dSpaceCollide ((dSpaceID)o2, data, &nearCallback);
      
!       } else {*/
      n = dCollide (o1,o2,N,&contact[0].geom,sizeof(dContact));
      if (n > 0) {
        for (i=0; i<n; i++) {
***************
*** 188,200 ****
        contact[i].surface.soft_erp = 0.5;
        contact[i].surface.soft_cfm = 0.3;
  	*/
! 	//dJointID c = dJointCreateContact (world,contactgroup,&contact[i]);
! 	dJointID c = dJointCreateContact (world,contactgroup,contact+i);
  	dJointAttach (c,
  		      dGeomGetBody(contact[i].geom.g1),
  		      dGeomGetBody(contact[i].geom.g2));
        }
!     }
    }
  
  }
--- 189,201 ----
        contact[i].surface.soft_erp = 0.5;
        contact[i].surface.soft_cfm = 0.3;
  	*/
! 	dJointID c = dJointCreateContact (world,contactgroup,&contact[i]);
! 	//dJointID c = dJointCreateContact (world,contactgroup,contact+i);
  	dJointAttach (c,
  		      dGeomGetBody(contact[i].geom.g1),
  		      dGeomGetBody(contact[i].geom.g2));
        }
!       //}
    }
  
  }
***************
*** 211,224 ****
    static float xyz[3] = {1.317f,-1.9817f,0.8000f};
    static float hpr[3] = {100.0000f,-27.5000f,0.0000f};
    dsSetViewpoint (xyz,hpr);
!   printf ("Press:\t'a' to increase speed.\n"
  	  "\t'z' to decrease speed.\n"
  	  "\t',' to steer left.\n"
  	  "\t'.' to steer right.\n"
  	  "\t' ' to reset speed and steering.\n"
  	  "\t'1' to save the current state to 'state.dif'.\n");
    // print the waypoint info
!   /*
    if (numWaypoints > 0)
      printf("waypoint change wp %i %f %f cp %f %f\n",wpcounter,
  	   waypoints[wpcounter].px,waypoints[wpcounter].py,
--- 212,226 ----
    static float xyz[3] = {1.317f,-1.9817f,0.8000f};
    static float hpr[3] = {100.0000f,-27.5000f,0.0000f};
    dsSetViewpoint (xyz,hpr);
!   /*
!     printf ("Press:\t'a' to increase speed.\n"
  	  "\t'z' to decrease speed.\n"
  	  "\t',' to steer left.\n"
  	  "\t'.' to steer right.\n"
  	  "\t' ' to reset speed and steering.\n"
  	  "\t'1' to save the current state to 'state.dif'.\n");
    // print the waypoint info
!   
    if (numWaypoints > 0)
      printf("waypoint change wp %i %f %f cp %f %f\n",wpcounter,
  	   waypoints[wpcounter].px,waypoints[wpcounter].py,
